crates/router/src/core/payments.rs

```diff
@@ -1280,16 +1283,16 @@ pub async fn list_payments(
                 .await
             {
                 Ok(pa) => Some(Ok((pi, pa))),
-                Err(e) => {
-                    if e.current_context().is_db_not_found() {
+                Err(error) => {
+                    if matches!(error.current_context(), StorageError::ValueNotFound(_)) {
```

Comment:
What about `StorageError::DatabaseError(Report<DatabaseError::NotFound>)`? As far as I remember, the `is_db_not_found()` function used to handle that as well.

Reply:
this is the StorageError defined in data_models it does not wrap around the database error type